### DISCUSSION TITLE
Long Wait Times to Save and Update Workspace Database

### DISCUSSION THREAD

**Discussion by campeauslalom** [TabularEditor/TabularEditor3#549] [🙋♀️TE3 Q&A] [answered]
My team has two models deployed to a Power BI workspace used as the development stage in a deployment pipeline. These models are largely identical aside from a slightly different security model, but are consistent in performance and the typical time it takes to update or save to the workspace DB. The workspace is on a premium capacity and set to Large Dataset Format.

The time it takes to update the workspace database when opening the model or save changes to the workspace database used to be extremely fast (typically less than 30 seconds) until roughly a month ago when it began to take significantly longer, generally in the range of 10-15 minutes. This is a consistent experience for everyone on our team that works with the models, and began happening to all of us at the same time. This has caused significant disruption to our workflow, as we've had to weigh the utility of connecting to a workspace DB vs. the time required to save changes vs. the risk of losing work due to crashes.

We've made many changes to the model over time, but none that stand out as particularly likely to increase the time required to save by this significant of a degree. To the best of everyone's recollection, this started happening out of the blue one day and was not preceded by any major structural changes. However, because we have other models using the same capacity that aren't experiencing similar issues, it certainly seems like something with the model itself that's to blame.

Is this something anyone else has experienced in the past, and if so was there some kind of structural feature that ended up being the primary cause? If this is something we can rectify we'd love to do that, but if it's something we're going to have to live with due to the complexity of the model we'd love to know that as well so we can explore making some changes to our workflow.

Thank you!

**Comment by otykier** [comment, marked as answer]
From your description, I'm afraid this is a Power BI service issue, but just to confirm, could you try to perform a deployment using TE2? That is, open the model in TE2, then go to Tools > Deploy... and deploy over your existing workspace database in the Power BI service (this is the same thing TE3 does when it's updating the workspace database).

Another thing to try, is to deploy the same model to a different Power BI workspace. If you're seeing any difference in saving/updating times, it indeed indicates a service issue, which should be reported to Microsoft.

**Reply by campeauslalom** [reply]
Thank you so much - I believe this is the answer. I deployed the model to a different workspace on the same capacity, and the save/update times are back around 30 seconds. We will report this issue to Microsoft as suggested. Many thanks for helping us troubleshoot this so quickly!

**Reply by campeauslalom** [reply]
As a follow-up to this for anyone it might help - since this was our dev environment we decided to delete and re-create the models in the workspace - this fixed the issues. This may not be a plausible solution for everyone, but for us it worked without causing any issues to our Power BI or CI/CD pipelines. Thanks again for the quick assistance!
